src/crypto/store/indexeddb-crypto-store-backend.js

```diff
@@ -368,13 +446,28 @@ function abortWithException(txn, e) {
     // We could alternatively make the thing we pass back to the app
     // an object containing the transaction and exception.
     txn._mx_abortexception = e;
-    txn.abort();
+    try {
+        txn.abort();
+    } catch (e) {
+        // sometimes we won't be able to abort the transaction
+        // (ie. if it's aborted or completed)
+    }
 }
 
 function promiseifyTxn(txn) {
     return new Promise((resolve, reject) => {
-        txn.oncomplete = resolve;
-        txn.onerror = reject;
+        txn.oncomplete = () => {
+            if (txn._mx_abortexception !== undefined) {
+                reject(txn._mx_abortexception);
+            }
+            resolve();
+        };
+        txn.onerror = () => {
+            if (txn._mx_abortexception !== undefined) {
```

Comment:
do we need this as *well* as the `onabort`?

Reply:
Since in `abortWithException`, we may not always be able to abort the txn, this is to cover the cases where we can't, so the exception still gets thrown.